core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -545,6 +551,13 @@ protected Scope visitRefreshMaterializedView(RefreshMaterializedView refreshMate
                         "Query: [" + Joiner.on(", ").join(queryTypes) + "]");
             }
 
+            analysis.setUpdateType(
+                    "REFRESH MATERIALIZED VIEW",
+                    targetTable,
+                    Optional.empty(),
+                    Optional.of(Streams.zip(insertColumns.stream(), queryTypes.stream(), (columnName, type) -> new Column(columnName, type.toString()))
```

Comment:
curious, is there a reason behind doing this differently than "INSERT" case?

Reply:
Nope. Have made it in sync with `INSERT` case.